zetaclient/evm_client.go

```diff
@@ -977,12 +998,14 @@ func (ob *EVMChainClient) observeERC20Deposited(startBlock, toBlock uint64) uint
 	// collect and sort events by block number, then tx index, then log index (ascending)
 	events := make([]*erc20custody.ERC20CustodyDeposited, 0)
 	for iter.Next() {
-		if !iter.Event.Raw.Removed && iter.Event.Raw.BlockNumber > 0 { // skip if chain reorg removed this event
+		// sanity check tx event
+		err := ob.CheckEvmTxLog(&iter.Event.Raw, addrCustody, "", TopicsDeposited)
+		if err == nil {
 			events = append(events, iter.Event)
 			continue
 		}
-		ob.logger.ExternalChainWatcher.Warn().Msgf("observeERC20Deposited: invalid event in tx %s at height %d for chain %d",
-			iter.Event.Raw.TxHash.Hex(), iter.Event.Raw.BlockNumber, ob.chain.ChainId)
+		ob.logger.ExternalChainWatcher.Error().Err(err).Msgf("observeERC20Deposited: invalid Deposited event in tx %s on chain %d at height %d",
```

Comment:
I would maybe keep a `Warn` log for these kind of error since it might be expected

Reply:
okay. I can set it back to `Warn` as DataDog still prints it out and we can use it for investigation.

Reply:
same [commit](https://github.com/zeta-chain/node/pull/1539/commits/cb257f672246dee79554c565f711dec5b3ab1345)